ansible/roles/mambaforge/tasks/mambaforge.yml

```diff
@@ -25,19 +17,37 @@
   failed_when: false
   register: conda_version_cmd
 
+- name: Wipe the existing installation if it differs from the target version
+  ansible.builtin.file:
+    path: "{{ mamba_home }}"
+    state: absent
+  when:
+    - conda_version_cmd['rc'] is defined
+    - conda_version_cmd['stdout'] is defined
+    - (conda_version_cmd['rc'] == 0) and (conda_version_cmd['stdout'] != 'conda ' ~ mambaforge_conda_version)
+
+- name: Verify that the installation directory doesn't exist
+  ansible.builtin.stat:
+    path: "{{ mamba_home }}"
+  register: mambaforge_home_check
+  failed_when: mambaforge_home_check['stat']['exists']
+
```

Comment:
Do we need this verification?
I'd assume that if the previous step's `state: absent` is not fulfilled the task already failed, no?
(Then again, I have to (re-) learn quite a bit of Ansible again, so might be missing somethings.)

Reply:
The point is to check that it's either a fresh installation or we wiped a previous installation. In a different situation (e.g. the directory is already present but contains something else, maybe due to a mistake in setting `mamba_home`) we fail to avoid damage.

Reply:
Regarding the `state: absent` step: it only runs if an existing installation needs to be wiped. If the directory exists but doesn't contain conda, it's skipped.